internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -400,10 +400,14 @@ func (h *PolicyChangeHandler) applyLoggingConfig(ctx context.Context, loggingCon
 	if loggingConfig != nil {
 		// we have logging config to set
 		policyLogLevel = &loggingConfig.Level
-	}
 
-	h.log.Infof("Setting fallback log level %v from policy", policyLogLevel)
-	return h.policyLogLevelSetter.SetLogLevel(ctx, policyLogLevel)
+		h.log.Infof("Setting fallback log level %v from policy", policyLogLevel)
+		return h.policyLogLevelSetter.SetLogLevel(ctx, policyLogLevel)
+	}
+	// use default log level
+	defaultLogLevel := logger.DefaultLogLevel
+	h.log.Infof("Setting fallback log level to default %s", defaultLogLevel)
+	return h.policyLogLevelSetter.SetLogLevel(ctx, &defaultLogLevel)
```

Comment:
If you set Info here, how do you clear a log level coming from policy to allow fallback onto the agent default level ? Having a nil log level value from the policy should be handled in the loglevel setter since that's the object that has the logic for proper priority and fallback, this will force the agent to always have a log level coming from policy even when there is non specified

Reply:
Before this change, when the policy log level was not set (nil), there wasn't any change made to the agent log level (that's why we had the bug on reassign).
This logic sets the fallback log level to info, which only applies if the agent doesn't have log level set.

I tested setting an agent log level to debug, and then making a policy change on a policy which has no log level. The agent in that case kept the debug log level, so I don't think there is any issue with this.

Reply:
Thanks for the explanation, I moved the logic to SetLogLevel, and tested that it works.